packaging/os/zypper.py

```diff
@@ -325,24 +341,56 @@ def package_absent(m, name):
 
     return retvals
 
+
+def repo_refresh(m):
+    "update the repositories"
+    retvals = {'rc': 0, 'stdout': '', 'stderr': '', 'changed': False, 'failed': False}
+
+    cmd = get_cmd(m, 'refresh')
+
+    retvals['cmd'] = cmd
+    result, retvals['rc'], retvals['stdout'], retvals['stderr'] = parse_zypper_xml(m, cmd)
+    if retvals['rc'] == 0:
+        # refresh was successed
+        retvals['changed'] = True
```

Comment:
changed means "package was installed/removed", so it should not be set here

Actually the `retvals['failed']` construct is overkill and you can just check `retvals['rc']` in the main function.